src/OpenTelemetry.Instrumentation.GrpcNetClient/Implementation/GrpcClientDiagnosticListener.cs

```diff
@@ -42,7 +42,7 @@ public GrpcClientDiagnosticListener(ActivitySourceAdapter activitySource, GrpcCl
 
         public override void OnStartActivity(Activity activity, object payload)
         {
-            if (!(this.startRequestFetcher.Fetch(payload) is HttpRequestMessage request))
+            if (!this.startRequestFetcher.TryFetch(payload, out HttpRequestMessage request))
```

Comment:
I think this should be `if (!this.startRequestFetcher.TryFetch(payload, out HttpRequestMessage request) || request == null)` to be equivalent to what it was before. Same comment on a few of them below this.

Reply:
just updated! Can you check if i missed anything?